triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -171,20 +186,112 @@ class Server(
   private def getTriggerStatus(uuid: UUID): Vector[(LocalDateTime, String)] =
     triggerLog.getOrDefault(uuid, Vector.empty)
 
-  private val route = concat(
+  // TODO[AH] Make sure this is bounded in size.
+  private val authCallbacks: TrieMap[UUID, Route] = TrieMap()
+
+  // This directive requires authorization for the given claims via the auth middleware, if configured.
+  // If no auth middleware is configured, then the request will proceed without attempting authorization.
+  //
+  // Authorization follows the steps defined in `triggers/service/authentication.md`.
+  // First asking for a token on the `/auth` endpoint and redirecting to `/login` if none was returned.
+  // If a login is required then this will store the current continuation in `authCallbacks`
+  // to proceed once the login flow completed and authentication succeeded.
+  private def authorize(claims: AuthRequest.Claims)(
```

Comment:
This directive seems reasonably generic and could potentially be factored out into a client library to be re-used between multiple clients that use the auth middleware.